### DISCUSSION TITLE
deepExtract Deprecation

### DISCUSSION THREAD

**Discussion by saggenerics** [renovatebot/renovate#13642] [Request Help] [answered]
Hello,

As deepExtract has been deprecated, will it be replaced by another function in the future and if so, what is your roadmap. In our repositories, we have been using Gradle for dependency resolving with Maps inside that specify the GAVs and it seemed like a good option to resolve these dependencies that are dynamically added using deepExtract.

Best Regards,
Kristiyan

**Comment by viceice** [comment]
The deep extract is already replaced by a js only implementation since some major versions, which you should use and report any issue about that. That's why you get the deprecation warning and an error in future.

So simply remove the deep and test.

**Comment by rarkins** [comment, marked as answer]
The deep extract will be removed shortly. For any feature gaps you'll need to either:
- Submit a feature request, with reproduction, etc, or
- Refactor how you define Gradle dependencies, such as using Gradle's new central version catalog if that's an option
